### PR TITLE
updated table code to remove need for app controller

### PR DESCRIPTION
**Pull request by MelSumner**
### :pushpin: Summary

If merged, this PR implements some changes that remove the need for an app controller. 

### :hammer_and_wrench: Detailed description

It no longer appends query params to the URL, but this is probably better for cases where there are multiple sortable tables on the same page. 

🚨 **HOWEVER, there's something about the change that is preventing tables from sorting correctly, need to troubleshoot more.**

Tests pass. 
table-tests

How to use docs updated: 
CleanShot 2022-10-13 at 12 21 45

### 👀 How to review

👉 Review by files changed

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/addon/components/hds/table/table-sortable.js [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
 
+    const currentColHeading = event.target;
+    const currColHeadingText = currentColHeading.textContent;
+
     const liveRegion = document.querySelector(`#${this.a11yMessageId}`);
-    liveRegion.textContent = `Sorted by ${column} ${this.args.sortOrder}ending`;
+    liveRegion.textContent = `Sorted by ${currColHeadingText} ${this.sortOrder}ending`;
     setTimeout(function () {
       liveRegion.textContent = '';
     }, 2000);
```
Comment (didoo) [inline packages/components/addon/components/hds/table/table-sortable.js:33]: [issue] you're overriding some of the changes already approved and merged in https://github.com/hashicorp/design-system/pull/625

see: https://github.com/hashicorp/design-system/blob/cb4c1554a175cbf77dd1d547f76a93c1d8e6e7de/packages/components/addon/components/hds/table/table-sortable.js#L13-L26

I think you need to merge `table-sortsupport` in this branch, or rebase this branch on `table-sortsupport` but avoid to bring in these changes (I can help you with this, if needed)

**Review by didoo — CHANGES_REQUESTED** [review]
Overall works for me. There is only one major issue that is that some of the existing code is being overwritten and should not.

Let me know once it's fixed, so I can approve it.

File: packages/components/addon/components/hds/table/table-sortable.js (hunk 1 above)
Comment (MelSumner) [reply packages/components/addon/components/hds/table/table-sortable.js:33]: this section isn't working at all, working on it now.

**Review by alex-ju — APPROVED** [review]
Looks good!

**Pull request merged**